onnxruntime/core/graph/graph_utils.h

```diff
@@ -39,16 +39,30 @@ bool IsOutputUsed(const Node& node, int index);
 bool IsGraphInput(const Graph& graph, const NodeArg* input);
 
 /** returns true if 'name' is an initializer, and is constant and cannot be overridden at runtime. 
-@param check_outer_scope If true and the graph is a subgraph, check parent graph/s for 'name' if not found in 'graph'.
+@param check_outer_scope If true and the graph is a subgraph, check ancestor graph/s for 'name' if not found in 'graph'.
 */
-bool IsConstantInitializer(const Graph& graph, const std::string& name, bool check_outer_scope);
+bool IsConstantInitializer(const Graph& graph, const std::string& name, bool check_outer_scope = true);
 
-/** Checks if the given node has only constant inputs (initializers). */
-bool AllNodeInputsAreConstant(const Graph& graph, const Node& node);
+/** returns the initializer's TensorProto if 'name' is an initializer, and is constant and 
+cannot be overridden at runtime. If the initializer is not found or is not constant a nullptr is returned.
+@param check_outer_scope If true and the graph is a subgraph, check ancestor graph/s for 'name' if not found in 'graph'.
+*/
+const ONNX_NAMESPACE::TensorProto* GetConstantInitializer(const Graph& graph, const std::string& name,
+                                                          bool check_outer_scope = true);
+
+/** Find the initializer called 'original_name' in 'graph', or its ancestors if check_outer_scope is true, 
+    and replace wtih 'initializer' in the correct graph.
+    Returns true if initializer was replaced. */
```

Comment:
I would add a note that this does not look into subgraphs of the graph when doing the replacement.